api/v1alpha1/flowcollector_types.go

```diff
@@ -201,6 +201,37 @@ type FlowCollectorKafka struct {
 	TLS ClientTLS `json:"tls"`
 }
 
+const (
+	PrometheusTLSDisabled = "DISABLED"
+	PrometheusTLSManual   = "MANUAL"
+	PrometheusTLSAuto     = "AUTO"
+)
+
+type PrometheusTLSConfigType string
+
+// PrometheusConfig define the prometheus endpoint configuration
+type PrometheusConfig struct {
+
+	//+kubebuilder:validation:Minimum=1
+	//+kubebuilder:validation:Maximum=65535
+	//+kubebuilder:default:=9102
+	// the prometheus HTTP port
+	Port int32 `json:"port,omitempty"`
+
+	// Select the type of TLS configuration
+	// "DISABLED" (default) to not configure TLS for the endpoint, "MANUAL" to manually provide cert file and a key file,
+	// and "AUTO" to use Openshift auto generated certificate using annotations
+	// +unionDiscriminator
```

Comment:
cc @mariomac is that how `unionDiscriminator` works? Looking at [your PR](https://github.com/netobserv/network-observability-operator/pull/144/files#diff-8b68f9600ecc25d88f615075d5135fca8f7e520634d6ae84027efdbc023b643a), I would think that the `enum` values need to match the discriminated field names; in other words, here, the enum for `MANUAL` would have to be `ManualTLS` ?
or this is pure coincidence in your code (with IPFIX/EBPF) and it is not required?

Reply:
PS I could find any doc about this `unionDiscriminator` annotation, do you have a link?

Reply:
According to the documentation [here](https://github.com/openshift/enhancements/blob/master/dev-guide/api-conventions.md#discriminated-unions)

> Discriminant values should be PascalCase and should be equivalent to the camelCase field name (json tag) of one member of the union

So having the enum and the field name matching is a recommendation, I will change the name here.

Reply:
yeah, in the PR @jotak  mention, it is done different,[ following the suggestion of an API review comment](https://github.com/openshift/api/pull/1218#discussion_r927636099), but double-checking the documentation examples, it seems the correct way is the way Olivier did.